web-library/src/main/java/ru/skypro/lessons/springboot/weblibrary/pojo/SecurityUserPrincipal.java

```diff
@@ -0,0 +1,59 @@
+package ru.skypro.lessons.springboot.weblibrary.pojo;
+
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+
+import java.util.Collection;
+
+public class SecurityUserPrincipal implements UserDetails {
+    private AuthUser user;
+
+    // Конструктор класса SecurityUserPrincipal,
+    // принимающий объект класса AuthUser.
+    public SecurityUserPrincipal(AuthUser user) {
+        this.user = user;
+    }
```

Comment:
Меняем поле и конструктор
```suggestion
    private User user;

    // Конструктор класса SecurityUserPrincipal,
    // принимающий объект класса AuthUser.
    public SecurityUserPrincipal(User user) {
        this.user = user;
    }
```